scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -0,0 +1,506 @@
+        try {
+            Multimap<String, String> groupsMap = TreeMultimap.create();
+            String line = null;
+            BufferedReader reader = new BufferedReader(new FileReader(groupFilePath));
+            try {
+                while ((line = reader.readLine()) != null) {
+                    if (!line.trim().isEmpty()) {
+                        String[] u2g = line.split(":");
+                        groupsMap.put(u2g[0].trim(), u2g[1].trim());
+                    }
+                }
+            } finally {
+                reader.close();
+            }
+            return groupsMap;
+        } catch (IOException e) {
+            exitWithErrorMessage("could not read group file : " + groupFilePath, null, e);
+        }
+        return null;
+    }
+
+    /**
+     * Stores the logins into
+     *
+     * @param loginFilePath
+     * @param props
+     * @throws IOException
+     */
+    private static void storeLoginFile(String loginFilePath, Properties props) throws IOException {
+        try (FileOutputStream stream = new FileOutputStream(new File(loginFilePath))) {
+            props.store(stream, null);
+        }
+        List<String> lines = null;
+
+        try (FileInputStream reader = new FileInputStream(loginFilePath)) {
+            lines = IOUtils.readLines(reader, Charsets.UTF_8);
+        }
+
+        TreeMap<String, String> sortedUsers = new TreeMap<>();
+        for (String line : lines) {
+            if (!(line.isEmpty() || line.startsWith("#"))) {
+                String[] loginAndPwd = line.split("=", 2);
+                sortedUsers.put(loginAndPwd[0], loginAndPwd[1]);
+            }
+        }
+        List<String> modifiedLines = new ArrayList<>(sortedUsers.size());
+        for (Map.Entry entry : sortedUsers.entrySet()) {
+            modifiedLines.add(entry.getKey() + ":" + entry.getValue());
+        }
+        try (FileOutputStream writer = new FileOutputStream(loginFilePath)) {
+            IOUtils.writeLines(modifiedLines, System.getProperty("line.separator"), writer, Charsets.UTF_8);
+        }
+    }
+
+    private static void storeGroups(String groupFilePath, Multimap<String, String> groups) throws ManageUsersException {
+        try {
+            try (PrintWriter writer = new PrintWriter(new FileWriter(groupFilePath))) {
+                for (String username : groups.keySet()) {
+                    for (String group : groups.get(username)) {
+                        writer.println(username + ":" + group);
+                    }
+                }
+            }
+
+        } catch (IOException e) {
+            exitWithErrorMessage("could not write group file : " + groupFilePath, null, e);
+        }
+    }
+
+    private static void exitWithErrorMessage(String errorMessage, String infoMessage, Throwable e) throws ManageUsersException {
+        throw new ManageUsersException(errorMessage, e, infoMessage);
+    }
+
+    private static String getLoginFilePath() {
```

Comment:
getLoginFilePath, getGroupFilePath, getPublicKeyFilePath seems very similar, maybe they can be refactored. something like : 

private static String getFilePath(String fileType) {
         if (!(new File(fileType).isAbsolute())) {
             //file path is relative, so we complete the path with the prefix Scheduler_Home constant
             fileType = PASchedulerProperties.SCHEDULER_HOME.getValueAsString() + File.separator + fileType;
         }
 
         return fileType;
     }